### DISCUSSION TITLE
Is there a way to colocate the translation files with my React components?

### DISCUSSION THREAD

**Discussion by npearson72** [i18next/i18next#2034] [Q&A] [answered]
Similar to css modules, tests, ect... I'd like to add my locale files next to my React components.

Imagine some deeply nested directory:

```
Foo.locale.json <-- Here my translation file is near my component
Foo.module.scss
Foo.text.tsx 
Foo.tsx 
```

This would be consistent with the general benefits of colocation, and would reduce the complexity of name spacing, key prefixes, general organizational concerns, ects.

Is this already possible but I've missed it in the docs? If not, has anyone jerry rigged their own similar solution?

**Comment by jamuhl** [comment, marked as answer]
not supported from our modules - but that request was also requested by others -> you should be able to find a solution in already closed issues: eg. https://github.com/i18next/react-i18next/issues/299#issuecomment-922892641

**Reply by jamuhl** [reply]
just to add - not supported by our modules - means just not supported without adding a little extra code by yourself

**Reply by npearson72** [reply]
Thanks for the reply
